### ISSUE TITLE
Please allow `.well-known/openid-configuration` configuration to be added while starting che via chectl

### ISSUE THREAD

**Issue by SDAdham** [eclipse-che/che#14604]
### Is your enhancement related to a problem? Please describe.

No

### Describe the solution you'd like

This can be already done via `global.cheDedicatedKeycloak=false,customOidcProvider=<oidc-url>,cheKeycloakClientId=<oidc_clientId>,customOidcUsernameClaim=<user_name_claim>` in https://www.eclipse.org/che/docs/che-6/kubernetes-multi-user.html#to-deploy-without-keycloak-in-multi-user-mode

also do we have an option to decide who can be the admin? i.e provide an email to be as admin?

### Describe alternatives you've considered

Considering to use helm as alternative for chectl for now?

### Additional context

I want to add gitlab as auth for che instead of keycloak, i can access `.well-known/openid-configuration` where the response is:
```
{"issuer":"https://myDomain.com","authorization_endpoint":"https://myDomain.com/oauth/authorize","token_endpoint":"https://myDomain.com/oauth/token","userinfo_endpoint":"https://myDomain.com/oauth/userinfo","jwks_uri":"https://myDomain.com/oauth/discovery/keys","scopes_supported":["api","read_user","read_repository","write_repository","sudo","openid","profile","email"],"response_types_supported":["code","token"],"response_modes_supported":["query","fragment"],"token_endpoint_auth_methods_supported":["client_secret_basic","client_secret_post"],"subject_types_supported":["public"],"id_token_signing_alg_values_supported":["RS256"],"claim_types_supported":["normal"],"claims_supported":["iss","sub","aud","exp","iat","sub_legacy","name","nickname","email","email_verified","website","profile","picture","groups"]}
```

**Comment by rhopp** [comment]
Nice catch. 
There is workaround, if you are blocked on this:
If you are using helm charts (default for chectl now), you can workaround this by cloning https://github.com/eclipse/che/tree/master/deploy, changing the helm charts and feeding them to chectl (by `--templates` parameter).

**Comment by SDAdham** [comment]
Hi @rhopp , I tried this:
```
helm upgrade --install che --namespace che -f ./values/tls.yaml --set global.ingressDomain=MyDomain.com,cheImage=eclipse/che-server:latest,global.cheDedicatedKeycloak=false,customOidcProvider=https://MyGitlab/.well-known/openid-configuration ./
```
But it didn't work, I ended up with single user installation, so I'm not sure if i'm missing anything

**Comment by SDAdham** [comment]
@rhopp : also, is there any plans for the custom admin to be provided as mentioned in the description?

**Comment by sleshchenko** [comment]
@SDAdham single user is a default installation, seems you should use 
```
helm upgrade --install che --namespace che -f ./values/tls.yaml -f ./values/multi-user.yaml --set global.ingressDomain=MyDomain.com,cheImage=eclipse/che-server:latest,global.cheDedicatedKeycloak=false,customOidcProvider=https://MyGitlab/.well-known/openid-configuration ./
```

**Comment by SDAdham** [comment]
Yes, I did, note: this documentation is incorrect >> https://www.eclipse.org/che/docs/che-6/kubernetes-multi-user.html#to-deploy-without-keycloak-in-multi-user-mode

the command should be without `/.well-known/openid-configuration` as per the logs of che's pod, it fails because it auto adds the rest of the link which is duplication, it ended up as `https://MyGitlab/.well-known/openid-configuration/.well-known/openid-configuration/`

It worked with this: `helm upgrade --install che --namespace che -f ./values/tls.yaml -f ./values/multi-user.yaml --set global.ingressDomain=MyDomain.com,cheImage=eclipse/che-server:latest,global.cheDedicatedKeycloak=false,customOidcProvider=https://MyGitlab ./`
But when i access che's url, I ended up with this: 
![image](https://user-images.githubusercontent.com/11182824/65685705-f66bdb00-e0a5-11e9-95b3-7e1a620b5f05.png)

**Comment by SDAdham** [comment]
From gitlab:
![image](https://user-images.githubusercontent.com/11182824/65685861-5793ae80-e0a6-11e9-9182-c3be9b728bc5.png)
Not sure, where do I provide appid and secret in che and what call back url should I be using?

**Comment by SDAdham** [comment]
I am trying:
```
gitHubClientID: ""
  gitHubClientSecret: ""
```
I think it could work if I can set `che.keycloak.github.endpoint=NULL`, see https://github.com/eclipse/che/blob/534a961e844049d5b4b84dd694176175eb367c97/assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/multiuser.properties#L135

**Comment by davidwindell** [comment]
@SDAdham I don't think you'll be able to use Gitlab until this issue is solved https://gitlab.com/gitlab-org/gitlab-foss/issues/48707#note_190990344

**Comment by SDAdham** [comment]
@davidwindell , I don't think so, gitlab is pretty much like github except that gitlab has tons of features, for github, we already take:
```
gitHubClientID: ""
  gitHubClientSecret: ""
```
These arguments are also available in gitlab, also with redirect_url, I'm confident if we allow those variable above to work as if they are gitlab a long with `che.keycloak.github.endpoint=NULL` to be provided by the user instead of auto-setting it during the installation, I am 100% sure it'll work.

**Comment by SDAdham** [comment]
@davidwindell : I assure you can verify via https://gitlab.com/profile/applications and https://gitlab.com/.well-known/openid-configuration

But please give the users the ability to set the server host, I integrated with many other applications that only request github but when I add gitlab, it works perfect!

**Comment by davidwindell** [comment]
@SDAdham did you ever get Che 7 to work wiith GitLab directly without Keycloak? I saw you raised https://github.com/eclipse/che-docs/pull/830

**Comment by SDAdham** [comment]
@davidwindell yes, I did

**Comment by SDAdham** [comment]
Allow integration with Gitlab auth rather than keycloak would be reaaaaaaaally nice to have. It should be really similar to Github. But we just need to specify the server for the gitlab

**Comment by cccs-eric** [comment]
Is it possible to re-open this issue?  I would also be very interested in using my OpenID server (in my case Azure AD).  I see `customOidcProvider` and `customOidcUsernameClaim`, but I don't see anything related to clientId/clientSecret for customOidc....  Am I missing something?
